src/main/java/net/earthcomputer/clientcommands/command/WaypointCommand.java

```diff
@@ -0,0 +1,413 @@
+        }
+
+        saveFile();
+        source.sendFeedback(Component.translatable("commands.cwaypoint.add.success", name, formatCoordinates(pos), dimension.location()));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int remove(FabricClientCommandSource source, String name) throws CommandSyntaxException {
+        String worldIdentifier = getWorldIdentifier(source.getClient());
+
+        Map<String, WaypointLocation> worldWaypoints = waypoints.get(worldIdentifier);
+
+        if (worldWaypoints == null) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        if (worldWaypoints.remove(name) == null) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        saveFile();
+        source.sendFeedback(Component.translatable("commands.cwaypoint.remove.success", name));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int edit(FabricClientCommandSource source, String name, BlockPos pos) throws CommandSyntaxException {
+        return edit(source, name, pos, source.getWorld().dimension());
+    }
+
+    private static int edit(FabricClientCommandSource source, String name, BlockPos pos, ResourceKey<Level> dimension) throws CommandSyntaxException {
+        String worldIdentifier = getWorldIdentifier(source.getClient());
+
+        Map<String, WaypointLocation> worldWaypoints = waypoints.get(worldIdentifier);
+
+        if (worldWaypoints == null) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        if (worldWaypoints.computeIfPresent(name, (key, value) -> new WaypointLocation(dimension, pos)) == null) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        saveFile();
+        source.sendFeedback(Component.translatable("commands.cwaypoint.edit.success", name, formatCoordinates(pos), dimension.location()));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int list(FabricClientCommandSource source) {
+        return list(source, false);
+    }
+
+    private static int list(FabricClientCommandSource source, boolean current) {
+        if (current) {
+            String worldIdentifier = getWorldIdentifier(source.getClient());
+
+            Map<String, WaypointLocation> worldWaypoints = waypoints.get(worldIdentifier);
+
+            if (worldWaypoints == null || worldWaypoints.isEmpty()) {
+                source.sendFeedback(Component.translatable("commands.cwaypoint.list.empty"));
+                return 0;
+            }
+
+            worldWaypoints.forEach((name, waypoint) -> source.sendFeedback(Component.translatable("commands.cwaypoint.list", name, formatCoordinates(waypoint.location()), waypoint.dimension().location())));
+            return worldWaypoints.size();
+        }
+
+        if (waypoints.isEmpty()) {
+            source.sendFeedback(Component.translatable("commands.cwaypoint.list.empty"));
+            return 0;
+        }
+
+        AtomicInteger count = new AtomicInteger();
```

Comment:
I don't really like using `AtomicInteger` for the purpose of mutating inside a lambda. You can either use Apache's `MutableInteger` or what I normally do is use a length-1 array (`int[] count = {0}; ... count[0] += worldWaypoints.size()`). Alternatively, you can use a for loop to iterate over the map's `entrySet()` to avoid the issue.